### PR TITLE
feat(analytics): Add command to migrate analytics data to pg

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [ ] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Add command to migrate feature analytics data from InfluxDB to Postgres

## How did you test this code?

Tested manually with an external InfluxDB and added a unit test.

### CODE REVIEW

File: api/app_analytics/management/commands/migrate_analytics.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    def add_arguments(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--migrate-till",
+            type=int,
+            dest="migrate_till",
+            help="Migrate data till n days ago",
+            default=30,
+        )
+
+    def handle(self, *args: Any, migrate_till: int, **options):
```
Comment (zachaysan) [inline api/app_analytics/management/commands/migrate_analytics.py]: nit: missing typing on `options` and the return type.

File: api/app_analytics/migrate_to_pg.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from app_analytics.constants import ANALYTICS_READ_BUCKET_SIZE
+from app_analytics.influxdb_wrapper import influxdb_client, read_bucket
+from app_analytics.models import FeatureEvaluationBucket
+
+
+def migrate_feature_evaluations(migrate_till: int = 30):
```
Comment (zachaysan) [inline api/app_analytics/migrate_to_pg.py]:
```suggestion
def migrate_feature_evaluations(migrate_till: int = 30) -> None:
```

File: api/tests/unit/app_analytics/test_migrate_to_pg.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from django.utils import timezone
+from pytest_mock import MockerFixture
+
+
+@pytest.mark.skipif(
+    "analytics" not in settings.DATABASES,
+    reason="Skip test if analytics database is not configured",
+)
+@pytest.mark.django_db(databases=["analytics", "default"])
+def test_migrate_feature_evaluations(mocker: MockerFixture):
```
Comment (zachaysan) [inline api/tests/unit/app_analytics/test_migrate_to_pg.py]:
```suggestion
def test_migrate_feature_evaluations(mocker: MockerFixture) -> None:
```

File: api/tests/unit/app_analytics/test_migrate_to_pg.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    # Next, mock the influx client and create some records
+    mock_influxdb_client = mocker.patch("app_analytics.migrate_to_pg.influxdb_client")
+    mock_query_api = mock_influxdb_client.query_api.return_value
+    mock_tables = []
+    for i in range(3):
+        mock_record = mocker.MagicMock(
+            values={"feature_id": feature_name, "environment_id": environment_id},
+            spec_set=["values", "get_time", "get_value"],
+        )
```
Comment (zachaysan) [inline api/tests/unit/app_analytics/test_migrate_to_pg.py:31]: TIL about `spec_set` that's really neat and useful!

File: api/tests/unit/app_analytics/test_migrate_to_pg.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+@pytest.mark.skipif(
+    "analytics" not in settings.DATABASES,
+    reason="Skip test if analytics database is not configured",
+)
+@pytest.mark.django_db(databases=["analytics", "default"])
+def test_migrate_feature_evaluations(mocker: MockerFixture):
+    # Given
+    feature_name = "test_feature_one"
+    environment_id = "1"
```
Comment (zachaysan) [inline api/tests/unit/app_analytics/test_migrate_to_pg.py:17]: It feels odd to me that `environment_id` is set to a string here.

**Review by zachaysan — APPROVED** [review]
Some minor nits, but overall looks great.

File: api/tests/unit/app_analytics/test_migrate_to_pg.py (hunk 5 above)
Comment (gagantrivedi) [reply api/tests/unit/app_analytics/test_migrate_to_pg.py:17]: Yeah, but unfortunately this is the data type our functions expects (historical reasons)

**Pull request merged**